signer/tests/regtest/mod.rs

```diff
@@ -0,0 +1,363 @@
+use bitcoincore_rpc::Auth;
+use bitcoincore_rpc::Client;
+use bitcoincore_rpc::Error as BtcRpcError;
+use bitcoincore_rpc::RpcApi;
+use secp256k1::Keypair;
+use secp256k1::Message;
+use secp256k1::SECP256K1;
+use std::sync::OnceLock;
+
+/// These must match the username and password in bitcoin.conf
+const BITCOIN_CORE_RPC_USERNAME: &str = "alice";
+const BITCOIN_CORE_RPC_PASSWORD: &str = "pw";
+
+pub const BITCOIN_CORE_FALLBACK_FEE: Amount = Amount::from_sat(1000);
+
+/// The name of our wallet on bitcoin-core
+const BITCOIN_CORE_WALLET_NAME: &str = "integration-tests-wallet";
+
+/// The faucet has a fixed secret key so that any mined amounts are
+/// preseved between test runs.
+const FAUCET_SECRET_KEY: &str = "0000000000000000000000000000000000000000000000000000000000000001";
+
+const FAUCET_LABEL: Option<&str> = Some("faucet");
+
+static BTC_CLIENT: OnceLock<(Client, Recipient)> = OnceLock::new();
+
+/// Initializes a blockchain and wallet on bitcoin-core. It can be called
+/// multiple times (even concurrently) but only generates the client and
+/// recipient once.
+///
+/// This function does the following:
+/// * Creates an RPC client to bitcoin-core
+/// * Creates or loads a watch-only wallet on bitcoin-core
+/// * Loads a "faucet" private-public key pair with a P2WPKH address.
+/// * Has the bitcoin-core wallet watch the generated address.
+/// * Ensures that the faucet has at least 1 bitcoin spent to its address.
+pub fn initialize_blockchain() -> &'static (Client, Recipient) {
+    BTC_CLIENT.get_or_init(|| {
+        let username = BITCOIN_CORE_RPC_USERNAME.to_string();
+        let password = BITCOIN_CORE_RPC_PASSWORD.to_string();
+        let auth = Auth::UserPass(username, password);
+        let rpc = Client::new("http://localhost:18443", auth).unwrap();
+
+        get_or_create_wallet(&rpc, BITCOIN_CORE_WALLET_NAME);
+        let faucet = Recipient::from_key(FAUCET_SECRET_KEY, AddressType::P2wpkh);
+        faucet.track_address(&rpc, FAUCET_LABEL);
+
+        let amount = rpc
+            .get_received_by_address(&faucet.address, Some(1))
+            .unwrap();
+
+        if amount < Amount::from_int_btc(1) {
+            faucet.generate_blocks(&rpc, 101);
+        }
+
+        (rpc, faucet)
+    })
+}
+
+fn get_or_create_wallet(rpc: &Client, wallet: &str) {
+    match rpc.load_wallet(wallet) {
+        // Success
+        Ok(_) => (),
+        // This happens if the wallet has already been loaded.
+        Err(BtcRpcError::JsonRpc(JsonRpcError::Rpc(RpcError { code: -35, .. }))) => (),
+        // The wallet probably hasn't been created yet, so lets do that
+        Err(_) => {
+            // We want a wallet that is watch only, since we manage keys
+            let disable_private_keys = Some(true);
+            rpc.create_wallet(wallet, disable_private_keys, None, None, None)
+                .unwrap();
+        }
+    };
+}
+
+/// Helper struct for representing an address on bitcoin.
```

Comment:
This structs represents more than just an address, since that is fully covered by the `Address` type. I assume the extra information is the fact that we have a key pair associated with this address, so I suggest updating the comment to something like

```suggestion
/// Helper struct for representing an address we control on bitcoin.
```